services/kvstore/src/main/java/org/hyperledger/besu/services/kvstore/SegmentedInMemoryKeyValueStorage.java

```diff
@@ -424,4 +424,30 @@ public void dump(final PrintStream ps) {
       lock.unlock();
     }
   }
+
+  /**
+   * Merge this storage's data to a SegmentedKeyValueStorageTransaction. This provides atomic
+   * merging of all data from this storage to the target transaction.
+   *
+   * @param otherTransaction the transaction to merge data into
+   */
+  public void mergeTo(final SegmentedKeyValueStorageTransaction otherTransaction) {
```

Comment:
I guess this is most probably safe, but are we sure that `otherTransaction` is thread safe here and does not need a `synchronized` block here?

Reply:
I don't think it's needed for the moment , only this part will change the transaction